planner/core/memtable_predicate_extractor.go

```diff
@@ -1408,32 +1410,80 @@ type StatementsSummaryExtractor struct {
 
 // Extract implements the MemTablePredicateExtractor Extract interface
 func (e *StatementsSummaryExtractor) Extract(
-	_ sessionctx.Context,
+	sctx sessionctx.Context,
 	schema *expression.Schema,
 	names []*types.FieldName,
 	predicates []expression.Expression,
 ) (remained []expression.Expression) {
 	// Extract the `digest` column
 	remained, skip, digests := e.extractCol(schema, names, predicates, "digest", false)
-	e.SkipRequest = skip
-	if e.SkipRequest {
+	if skip {
+		e.SkipRequest = true
 		return nil
 	}
 	if digests.Count() > 0 {
 		e.Enable = true
 		e.Digests = digests
 	}
+	// Extract time range
+	// SELECT ... WHERE summary_begin_time <= endTime AND summary_end_time >= startTime
+	tz := sctx.GetSessionVars().StmtCtx.TimeZone
+	remained, _, endTime := e.extractTimeRange(sctx, schema, names, remained, "summary_begin_time", tz)
+	remained, startTime, _ := e.extractTimeRange(sctx, schema, names, remained, "summary_end_time", tz)
```

Comment:
Is that alright to ignore `someTime <= summary_begin_time` and `someTime >= summary_end_time`?

Reply:
What we want to find is `[summary_begin_time, summary_end_time)` that has an intersection with `[begin, end)`, which is a comparison between time range and time range, not a comparison between time point and time range.

So the query we expect is:

```sql
SELECT * FROM `statements_summary_history` WHERE `summary_begin_time` <= `end` AND `summary_end_time` >= `begin`;
```

The corresponding figure is:

```
                               begin                 end
                                 |--------------------| 
                                 |                    |
       |-----------------------------------------|---------------------------------------|
summary_begin_time1   window1   summary_end_time1/summary_begin_time2   window2   summary_end_time2
```

What we want to find is the window that intersects with `[begin, end)`, it should be `{window1, window2}`.

Reply:
So what if meeting unexpected queries, i.e. queries don't match:
```sql
SELECT * FROM `statements_summary_history` WHERE `summary_begin_time` <= `end` AND `summary_end_time` >= `begin`;
```

Reply:
It will return all records. It will return all records. Of course, I think we can also return empty when the where clause does not meet expectations.

Reply:
https://github.com/pingcap/tidb/pull/40814/commits/34ff06bb45ef5ab875e6ee88269d48c7c9a0455b

After this commit, it should return the expected results although the time range is not as expected.